src/parsec.rs

```diff
@@ -151,38 +163,51 @@ impl ParsecMap {
         response: Response,
         pub_id: id::PublicId,
         log_ident: &LogIdent,
-    ) -> bool {
+    ) {
         // Increase the size before fetching the parsec to satisfy the borrow checker
         let ser_size = if let Ok(size) = bincode::serialized_size(&response) {
             size
         } else {
-            return false;
+            return;
         };
         self.count_size(ser_size, msg_version, log_ident);
 
+        trace!(
+            "{} - handle parsec response v{} from {} (size {}b)",
+            log_ident,
+            msg_version,
+            pub_id,
+            ser_size,
+        );
+
+        let _ = self.pending_requests.remove(pub_id.name());
+
         let parsec = if let Some(parsec) = self.map.get_mut(&msg_version) {
             parsec
         } else {
-            return false;
+            return;
         };
 
         if let Err(err) = parsec.handle_response(&pub_id, response) {
             debug!("{} - Error handling parsec response: {:?}", log_ident, err);
         }
-
-        self.last_version() == msg_version
     }
 
     pub fn create_gossip(
         &mut self,
         version: u64,
         target: &id::PublicId,
     ) -> Result<Variant, CreateGossipError> {
+        if !self.pending_requests.insert(*target.name()) {
```

Comment:
This actually means we cannot have more than one pending request to the same target. right?
In case of that, shall the returned error be handled to pick another candidate?

And, at https://github.com/maidsafe/routing/pull/2054/files#diff-98b4a6ae0c53dc7ababf2d21eb815263R264
seems when choosing the candidate, we already carried out filtering, so this case shall not be happen?